server/server.js

```diff
@@ -252,22 +261,13 @@ app.use('/genotet', function(req, res, next) {
         next();
       } else {
         log.serverLog(result.error);
-        jsonResponse(result, req, res);
+        jsonResponse(/** @type {Object} */(result), req, res);
```

Comment:
In fact, that user.findUsername calls database.getOne, of which the callback is function(Object).
https://github.com/ViDA-NYU/genotet/blob/a60493079bb4db8c15719f629c20983d3ac9ac8d/server/database.js#L42

Your parsed the username in function(Object), which gives you a string. Then you use the string to callback here. Thus it seems that result must be a string in this case.